bedrock/firefox/views.py

```diff
@@ -789,13 +789,14 @@ def get_context_data(self, **kwargs):
     def get_template_names(self):
         locale = l10n_utils.get_locale(self.request)
         variant = self.request.GET.get('v', None)
+        geo = self.request.GET.get('geo', None)
 
         # ensure variant matches pre-defined value
         if variant not in self.variations:
             variant = None
 
-        if locale == 'ru' and switch('firefox-yandex'):
-            template = 'firefox/new/trailhead/download-yandex.html'
+        if locale == 'ru' and geo == 'ru' and switch('firefox-yandex'):
```

Comment:
@pmac I could use some direction here for how to do the geolocation server side. We don't need to redirect, just use a different template so I wasn't sure how to use the GeoRedirectView or even if it's possible to use it together with a L10nTemplateView.

Reply:
The issue is that for geo to work on the server side we have to turn off the CDN cache because all requests have to make it to the server since we don't know their country of origin until it reaches the server. That's why we've restricted server side geo to redirects so that we don't have to turn off the cache for an HTML page so that it stays fast. What we'd normally do is to use the redirect view and have two other URLs that would load the template you want. Could even just be the same URL but with a query param.

Reply:
I don't think that's a good idea for the download page. I guess it's back to client side detection.